### ISSUE TITLE
Error with table size not being divisible by text size

### ISSUE THREAD

**Issue by jinyongyoo** [google-research/deduplicate-text-datasets#11]
Hi, I'm getting an error because size of the suffix table is not divisible by length of the text.
https://github.com/google-research/deduplicate-text-datasets/blob/8a172b0d815862b1131da203be24d430e121a725/src/main.rs#L479

I'm running `bash scripts/deduplicate_single_file.sh test.txt test_dedup.txt 20 1` where `test.txt` just contains a few paragraphs from a random Wikipedia article and some duplicate text that I manually added. I'm doing this mainly for debugging purpose (I would like to later make some edits to keep the first occurrence of duplicate samples and throw away the rest). If I run the command on my actual dataset that is roughly ~70GB big, I'm not encountering such issue. So I'm wondering what the issue is? Does the code not work with datasets that are too small? 

Thanks!

Update: I just found out that running the command on the actual 70GB dataset also raised the same error.

**Comment by carlini** [comment]
Huh. Thanks for reporting this. I'm not sure what's going on right now but I'll look into it later this week.

To help me debug, could you upload (assuming it's safe to do so, but because you're saying it's wikipedia data I suspect it is) the exact file you're using in case that makes a difference?

**Comment by jinyongyoo** [comment]
Yes absolutely! Here is the file [test.txt](https://github.com/google-research/deduplicate-text-datasets/files/8487819/test.txt) (note: it's very small)  I just copied the first few paragraphs of random articles (hint: they were Youtube, Google, and American Revolutionary War). Then I took line5 and line16 and duplicated them.

I ran this command `bash scripts/deduplicate_single_file.sh test.txt test_dedup.txt 20 1` and got the following output:
```
./target/debug/dedup_dataset make-part --data-file test.txt --start-byte 0 --end-byte 10490
Waiting for jobs to finish
Checking all wrote correctly
FACT 2.0
Rerunning 0 jobs because they failed.
Merging suffix trees
./target/debug/dedup_dataset merge --output-file tmp/out.table.bin --suffix-path test.txt.part.0-10490 --num-threads 40
Now merging individual tables
Cleaning up
    Finished dev [optimized + debuginfo] target(s) in 0.06s
     Running `target/debug/dedup_dataset self-similar --data-file test.txt --length-threshold 20 --cache-dir /tmp/cache --num-threads 1`
Start load!
thread 'main' panicked at 'assertion failed: metadata.len() % (text.len() as u64) == 0', src/main.rs:479:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    Finished dev [optimized + debuginfo] target(s) in 0.03s
     Running `target/debug/dedup_dataset collect --data-file test.txt --cache-dir /tmp/cache --length-threshold 20`
thread 'main' panicked at 'assertion failed: size_table % size_text == 0', src/main.rs:1049:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by jinyongyoo** [comment]
I also have an unrelated question: Can this program handle UTF-8 text (e.g. Korean)?

> Second, we don't want UTF8 strings. Everything is a [u8] byte array, because we might be working over token sequences which aren't valid UTF8.

I wasn't sure what this meant in the README.md. If we were to pass a file with utf-8 strings as input, does this mean we are essentially ignoring the utf-8 encoding and treat each byte individually for substring match?

**Comment by carlini** [comment]
Yeah, you're right in how it handles things: it doesn't understand the idea of a "letter" or "token" or anything -- it just thinks in terms of "bytes".

**Comment by carlini** [comment]
Okay this is very weird. I can't reproduce the error you see: when I run the commands everything comes out as expected.

Could you re-run the command and upload the following files:

test.txt.part.0-10490
test.txt.part.0-10490.table.bin
test.txt.table.bin

**Comment by jinyongyoo** [comment]
I think I figured out what the issue was. When I cleared the `tmp` folder and ran the command again, the error went away. My guess is that b/c the tmp folder never got cleared, I was merging tables from previous runs too. Maybe we shouldn't have this line commented out? https://github.com/google-research/deduplicate-text-datasets/blob/8a172b0d815862b1131da203be24d430e121a725/scripts/make_suffix_array.py#L90
